### PR TITLE
fix(platform): table drag and drop

### LINKED ISSUE #8602: [sourcing]: fdp-table - Drag and drop is not working properly
#### Is this a bug, enhancement, or feature request?
Bug
#### Briefly describe your proposal.
- Unable to drag before or after a row. It always drag into.
- Unable to drag a last row to the first row as the table does not scroll.
- When drag and drop, different newIndex or previousIndex are found when the parent expanded or not expanded. This is not correct. It should give the same index regardless of the parent’s open state or not.
- It freezes when drag a row inside a parent out and put on the top of the table.
#### If this is a bug, please provide steps for reproducing it.
- Create a table with parent/child rows
- Drag and drop
- If drag to a child, it will insert into
- If expand the parent, newIndex is different if parent is not expanded
#### Is there anything else we should know?
Had a meeting with Frank, Kevin and Manju

### PR DESCRIPTION
**Pull request by skaquib123**
## Related Issue(s)

closes #8602 

## Description

-Able to draw before and after a row. It is not going inside.
-Able to draw last row to first row and vice versa.

##### During Implementation

1. Visual Testing:

-   [x] visual misalignments/updates
-   [x] check Light/Dark/HCB/HCW themes
-   [x] RTL/LTR - proper rendering and labeling
-   [x] responsiveness(resize)
-   [x] Content Density (Cozy/Compact/(Condensed))
-   [x] States - hover/disabled/focused/active/on click/selected/selected hover/press state
-   [x] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
-   [x] Mouse vs. Keyboard support
-   [x] Text Truncation

2. API and functional correctness

-   [x] check for console logs (warnings, errors)
-   [x] API boundary values
-   [x] different combinations of components - free style
-   [x] change the API values during testing

3. Documentation and Example validations

-   [x] missing API documentation or it is not understandable
-   [x] poor examples
-   [x] Stackblitz works for all examples

4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

##### PR Quality

-   [x] the commit message(s) follows the guideline:
        https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
-   [x] tests for the changes that have been done
-   [x] all items on the PR Review Checklist are addressed :
        https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
-   [x] Run npm run build-pack-library and test in external application
-   [x] update `README.md`
-   [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)

### CODE REVIEW

**Review by platon-rov — COMMENTED** [review]
Although this is nice to have feature, it may be better to have both modes - place before/after and place inside, so we have to somehow distinct between those two.

**Comment by fkolar** [comment]
> Although this is nice to have feature, it may be better to have both modes - place before/after and place inside, so we have to somehow distinct between those two.

I agree. Let's move to this direction and maybe to unblock anybody who is using it and needs this so badly, plus currently see this as blocker or defect (that you can't simply drag now at all  - as now its limited).

 Let's take this as a part I. and the D&D "into & out" address  in the next PR. I am sure there will be other cases as well

File: libs/platform/src/lib/table/table.component.spec.ts [hunk 1]
```diff
@@ -1201,8 +1201,8 @@ class TreeTableDataProviderMock extends TableDataProvider<SourceTreeItem> {
                 calculateTableElementsMetaData();
```
Comment (fkolar) [inline libs/platform/src/lib/table/table.component.spec.ts:1202]: My only comment to this PR, would be add more unit tests and clearly validate before and after, first to last, last to first, last in between, first in between works correctly. I don't see any added tests here.

**Review by fkolar — CHANGES_REQUESTED** [review]

**Review by mikerodonnell89 — COMMENTED** [review]
Not unique to this PR, but when I drag Laptops here, why does Screens get expanded?

https://user-images.githubusercontent.com/2471874/199282575-181e23aa-0997-4608-ba4b-7f2efe709d33.mov

**Comment by skaquib123** [comment]
@mikerodonnell89 That feature was already existing that of dropped row will get expanded so i did not change it. Let me know if that also need to be changed.

**Comment by mikerodonnell89** [comment]
I am now only seeing changes to a test file

**Comment by skaquib123** [comment]
Hi @mikerodonnell89 ,  Can you please check in the commits and let me know if you can see?
link:
[https://github.com/SAP/fundamental-ngx/pull/8830/commits/6a4df64dd428900140fb17f806e0287b0aa41416](url)

**Pull request merged**